src/Exchange/ExchangerExchange.php

```diff
@@ -36,6 +40,10 @@ public function quote(Currency $baseCurrency, Currency $counterCurrency): Curren
             throw UnresolvableCurrencyPairException::createFromCurrencies($baseCurrency, $counterCurrency);
         }
 
-        return new CurrencyPair($baseCurrency, $counterCurrency, (string) $rate->getValue());
+        $rateValue = sprintf('%.14F', $rate->getValue());
+
+        assert(is_numeric($rateValue));
```

Comment:
Why is this `assert` necessary? I read on PHP manual page that from PHP 8 `sprintf` always returns string.

Reply:
Sorry, still don't get it. Why would you want it to be type checked when the only possible result is string? Why the `is_numeric` hint?

Reply:
In order for this `string` to be acceptable by endpoints that use it as a number, it must be a `numeric-string`.

`string` won't be sufficient.

Reply:
Ok, clear.